src/contract.ts

```diff
@@ -103,29 +103,30 @@ export class AllowedListContract {
   public getAssetIds = async (): Promise<string[] | Error> => {
     return await this.bridgeContract.methods.getAssetIds().call();
   };
-  
+
   public getTokenRegistryAllowedList = async (): Promise<AssetDetails[] | Error> => {
     // no filtering for now
     const assets = await this.getAssetIds();
     if (assets instanceof Error) return assets;
-  
+
     let assetsDetails: AssetDetails[] = [];
     for (let id of assets) {
       try {
         const details = await this.bridgeContract.methods.tokenRegistry(id).call();
-  
+
         if (details instanceof Error) return details;
-  
-        id = isHexStrict(id) ? id.substring(2) : id; // if 0x is there, then remove it
+
+        // id = isHexStrict(id) ? id.substring(2) : id; // if 0x is there, then remove it
         assetsDetails.push({
-          id: id,
-          min: fromWei(details.minimumValue),
+          id: stripHexPrefix(id), // if 0x is there, then remove it
+          min: fromWei(details.minimumValue, "microether"), // gives back microether = lovelace (for main asset),
+          // TODO: this will affect also other tokens/assets, but is this correct for them?
```

Comment:
Is it okay to merge with this TODO?

Reply:
this is only going to work for `wAda` not for any other assets. This is why there is this comment.

Reply:
We need to check the details's data to check if this is `WMAIN` or something else. @bartfish can you do that please ?

Reply:
sure